core/include/kernel/pseudo_ta.h

```diff
@@ -60,5 +62,44 @@ static inline struct pseudo_ta_ctx *to_pseudo_ta_ctx(struct ts_ctx *ctx)
 TEE_Result tee_ta_init_pseudo_ta_session(const TEE_UUID *uuid,
 			struct tee_ta_session *s);
 
+/*
+ * Helper functions for PTAs to support calls from a TA when CFG_PAN=y
+ */
+
+static inline bool is_caller_ta_with_pan(void)
+{
+	struct ts_session *s = NULL;
+	struct user_mode_ctx *uctx = NULL;
+
+	if (!IS_ENABLED(CFG_PAN))
+		return false;
+	s = ts_get_calling_session();
+	if (!s)
```

Comment:
The rest of this function can be replaced with `return s && is_user_ta_ctx(s->ctx)`;

Reply:
Done